### DISCUSSION TITLE
Merging Bounding Boxes for Multi-line Text

### DISCUSSION THREAD

**Discussion by Muhammadraafat1** [pymupdf/PyMuPDF#3688] [Looking for help]
Hello, When trying to obtain a single bounding box for a phrase that spans multiple lines. Specifically, the phrase "combination of hematoma" appears as follows:

"combination" is at the end of line 1.
"of hematoma" is at the beginning of line 2.

The corresponding bounding box information for each part of the phrase is as follows:

For "combination": (443.0003356933594, 351.9405212402344, 504.0020446777344, 361.21868896484375)
For "of hematoma": (87.00006103515625, 363.95977783203125, 150.83489990234375, 373.5115661621094)

When I print the text content using page.get_text().splitlines(), the output is:
['from the areolar border. This measured about 2 cm in size and probably is a combination ',
'of hematoma and a breast mass. There were no other masses in the breast and there was ']

I need to calculate a single bounding box that encompasses the entire phrase "combination of hematoma" despite it being split across two lines. My initial approach was to merge the bounding boxes: 
x1 = 443.0003356933594
y1 = 351.9405212402344
x2 = 150.83489990234375
y2 = 373.5115661621094
merged_bbox = (x1, y1, x2, y2)
but it's not work right

**Comment by JorjMcKie** [comment]
What you are trying to do **_cannot_** work!
When joining two rectangles, the result always is again a rectangle.
And for a non-empty rectangle we **_must_** have x1 < x2 **_and_** y1 < y2. I hope you see: you clearly violate this condition.

**Reply by Muhammadraafat1** [reply]
Ok, I think I need some explanation of your reply, I want to start of rectangle at "combination" and it's the end word in the first line so how bbox of it be smaller than bbox of start word in second line!
![image](https://github.com/user-attachments/assets/6e993deb-88d4-41af-a72e-c00f65e283b7)

**Reply by JorjMcKie** [reply]
There is no such thing as "single bounding box that encompasses the entire phrase" - and there cannot be one.
This is so utterly obvious that I am wondering whether we are talking about the same thing here.

**Reply by JorjMcKie** [reply]
In an effort to understand you: Do you actually want code that extracts the text in those 2 bounding boxes and no text else?

If you call your two bounding boxes bb1 and bb2, you can do this:
```python
print(page.get_text(clip=bb1))
print(page.get_text(clip=bb2))
```

**Reply by Muhammadraafat1** [reply]
No, I want when the output of page.add_redact_annot(rect, fill=(1, 1, 1)) is add a redaction annotation on "combination of hematoma" and those not in the same line as I mentioned above, Is that possible?

**Reply by JorjMcKie** [reply]
No!
You have to use 2 redaction of course, one for each bbox
